connect/src/main/java/com/stripe/android/connect/EmbeddedComponentManager.kt

```diff
@@ -1,67 +1,35 @@
 package com.stripe.android.connect
 
-import android.Manifest
-import android.app.Activity
-import android.app.Application.ActivityLifecycleCallbacks
 import android.content.Context
-import android.content.Intent
-import android.content.pm.PackageManager
-import android.net.Uri
-import android.os.Bundle
 import android.os.Parcelable
 import androidx.activity.ComponentActivity
-import androidx.activity.result.ActivityResultLauncher
-import androidx.activity.result.contract.ActivityResultContracts
 import androidx.annotation.RestrictTo
-import androidx.annotation.VisibleForTesting
-import androidx.core.content.ContextCompat.checkSelfPermission
 import androidx.fragment.app.FragmentActivity
-import com.stripe.android.connect.analytics.ComponentAnalyticsService
-import com.stripe.android.connect.analytics.ConnectAnalyticsService
 import com.stripe.android.connect.appearance.Appearance
 import com.stripe.android.connect.appearance.fonts.CustomFontSource
 import com.stripe.android.connect.di.StripeConnectComponent
-import com.stripe.android.connect.util.findActivity
-import com.stripe.android.connect.webview.ChooseFileActivityResultContract
-import com.stripe.android.connect.webview.serialization.ConnectInstanceJs
-import com.stripe.android.connect.webview.serialization.toJs
-import com.stripe.android.core.Logger
-import com.stripe.android.financialconnections.FinancialConnectionsSheet
-import com.stripe.android.financialconnections.FinancialConnectionsSheetResult
-import kotlinx.coroutines.flow.MutableSharedFlow
-import kotlinx.coroutines.flow.MutableStateFlow
-import kotlinx.coroutines.flow.StateFlow
-import kotlinx.coroutines.flow.asStateFlow
-import kotlinx.coroutines.flow.first
-import kotlinx.coroutines.suspendCancellableCoroutine
+import com.stripe.android.connect.manager.EmbeddedComponentCoordinator
 import kotlinx.parcelize.Parcelize
-import javax.inject.Inject
-import kotlin.coroutines.resume
 
 @PrivateBetaConnectSDK
 @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
-@SuppressWarnings("TooManyFunctions")
 class EmbeddedComponentManager(
-    private val configuration: Configuration,
-    private val fetchClientSecretCallback: FetchClientSecretCallback,
+    configuration: Configuration,
+    fetchClientSecretCallback: FetchClientSecretCallback,
     appearance: Appearance = Appearance(),
-    private val customFonts: List<CustomFontSource> = emptyList(),
+    customFonts: List<CustomFontSource> = emptyList(),
 ) {
-
-    private val _appearanceFlow = MutableStateFlow(appearance)
```

Comment:
👀 This _looks_ like a lot of changes, but it's almost entirely moving implementation details into `EmbeddedComponentCoordinator` and delegating where necessary.